flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/scala/expressionDsl.scala

```diff
@@ -262,15 +264,22 @@ trait ImplicitExpressionOperations {
   def collect: Expression = call(COLLECT, expr)
 
   /**
-    * Converts a value to a given type.
+    * Converts a value to a given data type.
     *
-    * e.g. "42".cast(Types.INT) leads to 42.
+    * e.g. "42".cast(DataTypes.INT()) leads to 42.
     *
     * @return casted expression
     */
-  def cast(toType: TypeInformation[_]): Expression =
+  def cast(toType: DataType): Expression =
     call(CAST, expr, typeLiteral(toType))
 
+  /**
+    * @deprecated Use [[cast(DataType)]] instead.
```

Comment:
nit: Could we add a bit longer note that it might be unsafe to mix the `DataType` and `TypeInformation` stacks?